dynaml/parser.go

```diff
@@ -4,8 +4,45 @@ import (
 	"container/list"
 	"strconv"
 	"strings"
+	
+	"github.com/cloudfoundry-incubator/spiff/yaml"
+	"github.com/cloudfoundry-incubator/spiff/debug"
 )
 
+/////////////////////////////////////////////////////////////////
+// internal helper nodes
+// used during expression parsing, they will never be contained in finally
+// parsed expression trees
+/////////////////////////////////////////////////////////////////
```

Comment:
We generally recommend not making explanatory block comments, especially for unexported, internal values. It is easy for these kinds of comments to become stale, for example on line 34, where it appears that the explanatory comment was copied and pasted from above and does not actually describe the helper struct below. 

We also believe that situations where code that is hard to understand without comments are opportunities to refactor into more human-readable code.

In this case we think the helper structs are fine as is, and do not need further commented explanation. Can you please remove the block comments separating these helpers here and below (line 42-44), as well as the explanatory block comments (line 25-27, 33-36).

Zak + @robdimsdale